lldb/source/Plugins/ExpressionParser/Swift/SwiftASTManipulator.cpp

```diff
@@ -122,87 +136,85 @@ __builtin_logger_initialize()
 
   StreamString wrapped_expr_text;
-  wrapped_expr_text.Printf("do\n"
-                           "{\n"
-                           "%s%s%s\n" // Don't indent the code so error columns
-                                      // match up with errors from compiler
-                           "}\n"
-                           "catch (let __lldb_tmp_error)\n"
-                           "{\n"
-                           "    var %s = __lldb_tmp_error\n"
-                           "}\n",
+
+  // Avoid indenting user code: this makes column information from compiler
+  // errors match up with what the user typed.
+  wrapped_expr_text.Printf(R"(
+do {
+%s%s%s
+} catch (let __lldb_tmp_error) {
+  var %s = __lldb_tmp_error
+}
+)",
                            GetUserCodeStartMarker(), text,
                            GetUserCodeEndMarker(), GetErrorName());
 
-  if (needs_object_ptr | static_method) {
+  if (needs_object_ptr || static_method) {
     const char *func_decorator = "";
     if (static_method) {
       if (is_class)
         func_decorator = "final class";
       else
         func_decorator = "static";
-    } else if (is_class &&
-               !(weak_self)) {
+    } else if (is_class && !weak_self) {
       func_decorator = "final";
     } else {
       func_decorator = "mutating";
     }
 
     const char *optional_extension =
-        (weak_self)
-            ? "Swift.Optional where Wrapped == "
-            : "";
-
-    wrapped_stream.Printf(
-        "extension %s$__lldb_context {\n"
-        "  @LLDBDebuggerFunction %s\n"
-        "  %s func $__lldb_wrapped_expr_%u(_ $__lldb_arg : "
-        "UnsafeMutablePointer<Any>) {\n"
-        "%s" // This is the expression text (with newlines).
-        "  }\n"
-        "}\n"
-        "%s\n"
-        "func $__lldb_expr(_ $__lldb_arg : UnsafeMutablePointer<Any>) {\n"
-        "  do {\n"
-        "    $__lldb_injected_self.$__lldb_wrapped_expr_%u(\n"
-        "      $__lldb_arg\n"
-        "    )\n"
-        "  }\n"
-        "}\n",
-        optional_extension, availability.c_str(), func_decorator,
-        current_counter, wrapped_expr_text.GetData(), availability.c_str(),
-        current_counter);
+        weak_self ? "Swift.Optional where Wrapped == " : "";
+
+    // The expression text is inserted into the body of $__lldb_wrapped_expr_%u.
+    wrapped_stream.Printf(R"(
+extension %s$__lldb_context {
+  @LLDBDebuggerFunction %s
+  %s func $__lldb_wrapped_expr_%u(_ $__lldb_arg : UnsafeMutablePointer<Any>) {
+    %s
+  }
+}
+%s
+func $__lldb_expr(_ $__lldb_arg : UnsafeMutablePointer<Any>) {
+  do {
+    $__lldb_injected_self.$__lldb_wrapped_expr_%u(
+      $__lldb_arg
+    )
+  }
+}
+)",
+                          optional_extension, availability.c_str(),
+                          func_decorator, current_counter,
+                          wrapped_expr_text.GetData(), availability.c_str(),
+                          current_counter);
 
     first_body_line = 5;
```

Comment:
I _think_ this was off by one before, but with the raw string literal change, it now looks correct.